x-pack/legacy/plugins/lens/server/routes/index_stats.ts

```diff
@@ -164,3 +275,170 @@ export function recursiveFlatten(
   });
   return returnTypes;
 }
+
+export async function getNumberHistogram(
+  aggSearchWithBody: (body: unknown) => Promise<unknown>,
+  field: { name: string; type: string; esTypes?: string[] }
+) {
+  const minMaxResult = (await aggSearchWithBody({
+    sample: {
+      sampler: { shard_size: 5000 },
+      aggs: {
+        min_value: {
+          min: { field: field.name },
+        },
+        max_value: {
+          max: { field: field.name },
+        },
+        top_values: {
+          terms: { field: field.name, size: 10 },
+        },
+      },
+    },
+  })) as AggregationSearchResponse<
+    unknown,
+    {
+      body: {
+        aggs: {
+          sample: {
+            sampler: {};
+            aggs: {
+              min_value: {
+                min: { field: string };
+              };
+              max_value: {
+                max: { field: string };
+              };
+            };
+          };
+        };
+      };
+    }
+  >;
+
+  const minValue = minMaxResult.aggregations!.sample.min_value.value;
```

Comment:
Just double checking: is it possible for aggregations to ever be null here?

Reply:
It should always return a body if the aggregation query ran at all, but maybe I'm missing an error handler somewhere around the request

Reply:
`aggregations` will be undefined if no matching indices were found. It's an edge case, but it happens, so it's reflected in the type.